proto/addressmetadata.proto

```diff
@@ -1,10 +1,34 @@
 syntax = "proto3";
 package models;
 
-// Basic key/value used to store header data.
-message Header {
-    string name = 1;
-    string value = 2;
+// Specifies the revocation method
+enum RevocationMethod {
+    /* If any of the pubkeys given in commitment_data[..n-1] signs
+    data commitment_data[n] then revocation is granted. */
+    BackupPubkey = 0;
+}
+
+// Message sent to server to revoke a key
+message Revocation {
+    // The revocation method
+    RevocationMethod method = 1;
+    // Data associated with the reveal phase
+    repeated bytes reveal_data = 2;
+    // New key to redirect to
+    bytes pub_key_hash = 3;
+    // Memo attached to revocation
+    string memo = 4;
+}
+
+// A RevocationClause defines conditions under which the a key can be revoked.
+message RevocationClause {
+    // The revocation method
+    RevocationMethod method = 1;
+    // Data associated with commitment phase
+    repeated bytes commitment_data = 2;
+    // Time period in which revocation is permitted
+    int64 valid_from = 3;
```

Comment:
I don't think this is correct.  This data (if relevant) would be included on a revocation message itself.

(e.g. you want to allow a second key to revoke yours that you're going to give to a third party.  You would produce a signed revocation certificate that includes this data and is signed by your key.  You'd then give that over to them.  The server (and clients which need to verify the revocation) would then accept it if it was valid.  However, this timing stuff wouldn't be relevant to other commitment schemes.)